packages/@aws-cdk/aws-s3/lib/notification-dest.ts

```diff
@@ -0,0 +1,37 @@
+import cdk = require('@aws-cdk/cdk');
+import { Bucket } from './bucket';
+
+/**
+ * Implemented by constructs that can be used as bucket notification targets.
+ */
+export interface INotificationDestination {
+    /**
+     * Registers this resource to receive notifications for the specified bucket.
+     * @param bucket The bucket. Use the `path` of the bucket as a unique ID.
+     */
+    bucketNotificationDestination(bucket: Bucket): NotificationDestinationProps;
```

Comment:
I thought we said the standard name for this function was going to be `notificationDestination` (based off the name of the interface and the props type).

Reply:
We did, but because of namespacing the interface name lost some of the context. However as a method of an L2 construct “notificationDestination” is not sufficient. We must have some delinataion as for where is this coming from. Otherwise it will look very odd. Maybe ‘s3NotificationDestination’?

Otherwise, I can rename the interface to ‘BucketNotificationDestination’. What do you think?

I am also wondering - ideally users shouldn’t care about these “inverted control” methods whatsoever (we now have a few of them). I wonder if we should find some convention that makes it intuitively clear that you don’t need to call this yourself.

We could prefix with something like an underscore... or something like “asBucketNotificationDestination”. 

I kinda like the “as” prefix. 

Thoughts?

Reply:
Okay, changed to `asBucketNotificationDestination`. It's a mouthful, but not really intended to be used by users, and I feel the `as` prefix works.